crypto/fipsmodule/sha/sha3.c

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017-2020 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal.h"
+#include <string.h>
+
+uint8_t *SHA3_256(const uint8_t *data, size_t len,
+                  uint8_t out[SHA3_256_DIGEST_LENGTH]) {
+  KECCAK1600_CTX ctx;
+  int ok = (SHA3_Init(&ctx, SHA3_PAD_CHAR, SHA3_256_DIGEST_BITLENGTH) && 
+            SHA3_Update(&ctx, data, len) &&
+            SHA3_Final(out, &ctx));
+
+  OPENSSL_cleanse(&ctx, sizeof(ctx));
+  if (ok == 0){
+    return NULL;
```

Comment:
is there a reason to return `NULL` here because I see that in sha1/2 implementation we always return `out`? Or should we also return `NULL` in sha1/2

Reply:
The return value check for SHA1/2 it is used for the `service_indicator`, however, since we don't check the success of the functions for SHA3 using the service indicator, I added NULL similar to [hmac.c](https://github.com/manastasova/aws-lc/blob/eca066dc173102306f2c1ea31d5eb8f671095dac/crypto/fipsmodule/hmac/hmac.c#L207). 
I saw that [md4](https://github.com/manastasova/aws-lc/blob/eca066dc173102306f2c1ea31d5eb8f671095dac/crypto/fipsmodule/md4/md4.c#L72) and [md5](https://github.com/manastasova/aws-lc/blob/eca066dc173102306f2c1ea31d5eb8f671095dac/crypto/fipsmodule/md5/md5.c#L74) are simply not checking the return values and always return out. I can change if needed but I thought that it is useful to check and return `NULL` instead of incorrect `out`.